### PR TITLE
 [#39455] Allow filtering of "empty" date fields (start/finish/custom)

### PR DESCRIPTION
**Pull request by mrmir**
https://community.openproject.org/work_packages/39455

### CODE REVIEW

File: spec/features/work_packages/is_empty_filter_spec.rb [hunk 1]
```diff
@@ -0,0 +1,111 @@
+#-- copyright
```
Comment (aaron-contreras) [inline spec/features/work_packages/is_empty_filter_spec.rb]:
```suggestion
# frozen_string_literal: true

#-- copyright
```

File: spec/features/work_packages/is_empty_filter_spec.rb [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+# See COPYRIGHT and LICENSE files for more details.
+#++
+
+require 'spec_helper'
+
+RSpec.describe 'Filter by date with "is empty"', js: true do
```
Comment (aaron-contreras) [inline spec/features/work_packages/is_empty_filter_spec.rb]: There's a new Rubocop well... - cop - that suggests the symbol style metadata. Also, using cuprite add a small performance boost to this spec out of the box :)

```suggestion
RSpec.describe 'Filter by date with "is empty"', :js, :with_cuprite do
```

File: spec/features/work_packages/is_empty_filter_spec.rb [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+  let(:work_package_with_custom_date) do
+    create(:work_package,
+           project:)
+  end
+
+  let(:custom_field_date) do
+    create(:date_wp_custom_field,
+           name: 'Date CF',
+           is_filter: true,
+           searchable: true).tap do |custom_field|
```
Comment (aaron-contreras) [inline spec/features/work_packages/is_empty_filter_spec.rb]: Recently learned that FactoryBot yields the instantiated object if a block is provided; no need to `#tap` :)
```suggestion
           searchable: true) do |custom_field|
```

**Review by aaron-contreras — CHANGES_REQUESTED** [review]
Nice! 🖌️ Left some suggestions in the specs.

I feel the feature spec better fits under `spec/features/work_packages/table/queries/is_empty_filter_spec.rb`. All other filter feature specs are grouped there so it would make it easier to discover if it were grouped there as well

File: spec/features/work_packages/is_empty_filter_spec.rb (hunk 2 above)
Comment (mrmir) [reply spec/features/work_packages/is_empty_filter_spec.rb]: Hey @aaron-contreras, what does this mean?

> There's a new Rubocop well... - cop - that suggests the symbol style metadata.

Comment (aaron-contreras) [reply spec/features/work_packages/is_empty_filter_spec.rb]: `js: true` can also be written as `:js`, as RSpec picks up the example group and infers you'd like to set it to true. The style that RSpec is enforcing now is the latter. It also reads a little more declaratively that way :)

Comment (mrmir) [reply spec/features/work_packages/is_empty_filter_spec.rb]: Aah okay, thanks!

**Comment by mrmir** [comment]
Thanks for those suggestions and for finding a better spot for the file @aaron-contreras, ready for you to review again :)

**Review by aaron-contreras — APPROVED** [review]

**Pull request merged**